data/icon-overlay.js

```diff
@@ -23,19 +23,21 @@ if ((host.indexOf('youtube.com') > -1) && ytHomeContainers.length) {
 }
 
 // YOUTUBE WATCH PAGE
-const ytWatchContainer = document.querySelector('.html5-video-player');
-if ((host.indexOf('youtube.com') > -1) && ytWatchContainer) {
-  ytWatchContainer.classList.add('minvid__overlay__wrapper');
-  const tmp = getTemplate();
-  tmp.addEventListener('click', function(ev) {
-    evNoop(ev);
-    self.port.emit('launch', {
-      url: window.location.href,
-      domain: host
+setInterval(() => {
```

Comment:
@kaganjd So, you might rearrange your patch to do something like:

- if you are on youtube.com, start the interval.
- inside the callback, if ytWatchContainer is truthy (querySelector will return null if the element isn't found), clear the interval, then do the rest of the stuff starting with adding the overlay__wrapper class